src/SourceConstraints.hs

```diff
@@ -88,13 +99,48 @@ data IEClass = Module String | Type String | Operator String | Function String
   deriving stock (Eq, Ord)
 
 locatedWarnings :: Data a => Context -> a -> WarningMessages
-locatedWarnings context node = listToBag $ catMaybes
-  [ mkQ empty sortedImportStatement  node
-  , mkQ empty sortedIEThingWith      node
-  , mkQ empty sortedIEs              node
-  , mkQ empty sortedMultipleDeriving node
-  ]
+locatedWarnings context@Context{..} node =
+  singleWarnings `unionBags` mkQ emptyBag absentImportDeclList node
   where
+    singleWarnings = listToBag $ catMaybes
+      [ mkQ empty sortedImportStatement  node
+      , mkQ empty sortedIEThingWith      node
+      , mkQ empty sortedIEs              node
+      , mkQ empty sortedMultipleDeriving node
+      ]
+
+    absentImportDeclList :: HsModule GhcPs -> WarningMessages
+    absentImportDeclList HsModule{..} =
+      listToBag $ catMaybes (absentList <$> candidates)
+      where
+        absentList :: LImportDecl GhcPs -> Maybe ErrMsg
+        absentList = \case
+          (L _loc ImportDecl { ideclHiding = Just (False, L loc list) }) ->
```

Comment:
@mbj Whitespace before and after `{` (and `}`) is inconsistent with other places this construct gets used.

Reply:
I think its canonical to have that whitespace. The other places (outside the diff) are at fault.

A future pass will fix this.

The only exception is `{..}`.